rust-runtime/aws-smithy-client/src/lib.rs

```diff
@@ -173,19 +175,19 @@ where
         let timeout_params =
             ClientTimeoutParams::new(&self.operation_timeout_config, self.sleep_impl.clone());
 
-        let svc = ServiceBuilder::new()
-            .layer(TimeoutLayer::new(timeout_params.operation_timeout))
-            .retry(
-                self.retry_policy
-                    .new_request_policy(self.sleep_impl.clone()),
-            )
-            .layer(TimeoutLayer::new(timeout_params.operation_attempt_timeout))
-            .layer(ParseResponseLayer::<O, Retry>::new())
-            // These layers can be considered as occurring in order. That is, first invoke the
-            // customer-provided middleware, then dispatch dispatch over the wire.
-            .layer(&self.middleware)
-            .layer(DispatchLayer::new())
-            .service(connector);
+        // Create a service by constructing it in reverse order. Think of this like an onion.
+        // The first service created will be the innermost service, wrapping the connector.
+        // The last service created will be the outermost service, wrapping all other services.
+        let svc = DispatchService::new(connector);
+        let svc = self.middleware.layer(svc);
+        let svc = ParseResponseService::<_, O, Retry>::new(svc);
+        let svc = TimeoutService::new(svc, timeout_params.operation_attempt_timeout);
+        let svc = RetryLayer::new(
```

Comment:
For my understanding, why is this using `RetryLayer` instead of `RetryService`?

Reply:
I tried to import `RetryService` from `tower` and couldn't find it so I assumed I was intended to use the layer to create it. I just looked at the `tower` source and they do have a `RetryService`, it's just called `Retry` instead. I'd switch over except I'm going to close this PR in favor of the orchestrator work I'm doing.